cmd/genji/shell/command.go

```diff
@@ -2,20 +2,94 @@ package shell
 
 import (
 	"fmt"
+	"os"
 	"sort"
+	"strings"
 
+	"github.com/agnivade/levenshtein"
 	"github.com/genjidb/genji"
 	"github.com/genjidb/genji/document"
 )
 
-var commands = map[string]string{
-	".tables":               "\t\tList names of tables.",
-	".exit":                 "\t\t\tExit this program.",
-	".indexes [table_name]": "\tDisplay all indexes or the indexes of the given table name.",
-	".help":                 "\t\t\tList all commands.",
+type Command struct {
+	Name    string
+	Options string
+	Help    string
+	Action  func(db *genji.DB, cmd []string) error
 }
 
-func runTablesCmd(db *genji.DB, cmd []string) error {
+type lessFunc func(a, b *Command) int
+
+type Commands struct {
+	cmds      []Command
+	less      []lessFunc
+	completed bool
+}
+
+var commands Commands
+
+func (c *Command) match(in string) bool {
+	return c.Name == in
+}
+
+func (c *Command) getHelp() string {
+	return c.Help
+}
+
+func (c *Command) getName() string {
+	return c.Name
+}
+
+func (c *Command) getOptions() string {
+	return c.Options
+}
+
+func (c *Command) Run(db *genji.DB, cmd []string) error {
+	commands.completed = false
+	return c.Action(db, cmd)
+}
+
+// Sort sorts the argument slice according to the less functions passed to OrderedBy.
+func (c *Commands) Sort(cmd []Command) {
+	c.cmds = cmd
+	sort.Sort(c)
+}
+
+// OrderedBy returns a Sorter that sorts using the less functions, in order.
+// Call its Sort method to sort the data.
+func OrderedBy(less ...lessFunc) *Commands {
+	return &Commands{
+		less: less,
+	}
+}
+
+// Len is part of sort.Interface.
```

Comment:
```suggestion
// Len implements sort.Interface.
```

Reply:
If we don't keep the sort I think is better to delete the `Len()` implementation.